src/backend/query_builder.rs

```diff
@@ -1177,6 +1198,70 @@ pub trait QueryBuilder: QuotedBuilder {
         }
     }
 
+    #[doc(hidden)]
+    /// Translate [`Frame`] into SQL statement.
+    fn prepare_frame(&self, frame: &Frame, sql: &mut SqlWriter, collector: &mut dyn FnMut(Value)) {
+        match *frame {
+            Frame::UboundedPreceding => write!(sql, " UNBOUNDED PRECEDING ").unwrap(),
+            Frame::Preceding(v) => {
+                self.prepare_value(&Some(v).into(), sql, collector);
+                write!(sql, " PRECEDING ").unwrap();
+            }
+            Frame::CurrentRow => write!(sql, " CURRENT ROW ").unwrap(),
+            Frame::Following(v) => {
+                self.prepare_value(&Some(v).into(), sql, collector);
+                write!(sql, " FOLLOWING ").unwrap();
+            }
+            Frame::UnboundedFollowing => write!(sql, " UNBOUNDED FOLLOWING ").unwrap(),
+        }
+    }
+
+    #[doc(hidden)]
+    /// Translate [`WindowStatement`] into SQL statement.
+    fn prepare_window_statement(
+        &self,
+        window: &WindowStatement,
+        sql: &mut SqlWriter,
+        collector: &mut dyn FnMut(Value),
+    ) {
+        let (first, exprs) = &window.partition_by;
+        write!(sql, " PARTITION BY ").unwrap();
+        self.prepare_simple_expr(first, sql, collector);
```

Comment:
From what I understand the `PARTITION BY` is not mandatory?
For instance, we can write
```sql
SELECT salary, sum(salary) OVER (ORDER BY salary) FROM empsalary;
```
Please correct me if I'm wrong

Reply:
Yes, you are right. Done